src/persist-txn/src/txn_write.rs

```diff
@@ -158,6 +147,18 @@ where
                     );
                     return Err(txns_upper);
                 }
+
+                // Validate that the involved data shards are all registered.
```

Comment:
It can turn the error above into a panic

```Rust
                if commit_ts < txns_upper {
                    debug!(
                        "commit_at {:?} mismatch current={:?}",
                        commit_ts, txns_upper
                    );
                    return Err(txns_upper);
                }
```

If `commit_ts < txns_upper` and the shard has been forgotten after `commit_ts` then we panic instead of returning an error. I was seeing this panic a bunch in the `stress_correctness` test.